pandas/core/indexes/datetimes.py

```diff
@@ -286,16 +281,16 @@ def __new__(cls, data=None,
             verify_integrity = True
 
         if data is None:
-            dtarr = DatetimeArray._generate_range(
-                start, end, periods,
-                freq=freq, tz=tz, normalize=normalize,
-                closed=closed, ambiguous=ambiguous)
             warnings.warn("Creating a DatetimeIndex by passing range "
                           "endpoints is deprecated.  Use "
                           "`pandas.date_range` instead.",
                           FutureWarning, stacklevel=2)
-
-            return cls(dtarr, name=name)
+            dtarr = DatetimeArray._generate_range(
+                start, end, periods,
```

Comment:
should the warning message actually be in DTA._generate_range?

Reply:
No, the non-deprecated way it gets called is via date_range